source/numerics.tex

```diff
@@ -6976,100 +6982,65 @@
 \begin{itemdescr}
 \pnum
 \requires The length of the array to which the argument refers equals \tcode{size()}.
+The value of an element in the left-hand side of a \tcode{valarray} assignment
+operator does not depend on the value of another element in that left-hand side.
 
 \pnum
 These operators allow the results of a generalized subscripting operation
 to be assigned directly to a
 \tcode{valarray}.
-
-\pnum
-If the value of an element in the left-hand side of a valarray assignment
-operator depends on the value of another element in that left-hand side,
-the behavior is undefined.
 \end{itemdescr}
 
 \rSec3[valarray.access]{\tcode{valarray} element access}
 
 \indexlibrarymember{operator[]}{valarray}%
 \begin{itemdecl}
-const T&  operator[](size_t) const;
-T& operator[](size_t);
+const T&  operator[](size_t n) const;
+T& operator[](size_t n);
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-The subscript operator
-returns a reference to the corresponding element of the array.
-
-\pnum
-Thus, the expression
-\tcode{(a[i] = q, a[i]) == q}
-evaluates as \tcode{true} for any non-constant
-\tcode{valarray<T> a},
-any
-\tcode{T q},
-and for any
-\tcode{size_t i}
-such that the value of
-\tcode{i}
-is less than the length of
-\tcode{a}.
+\requires
+\tcode{n < size()}.
 
 \pnum
-The expression
-\tcode{\&a[i+j] == \&a[i] + j}
-evaluates as \tcode{true} for all
-\tcode{size_t i}
-and
-\tcode{size_t j}
-such that
-\tcode{i+j}
-is less than the length of the array
-\tcode{a}.
+\returns
+A reference to the corresponding element of the array.
+\begin{note}
+The expression \tcode{(a[i] = q, a[i]) == q}
+evaluates as \tcode{true} for any non-constant \tcode{valarray<T> a},
+any \tcode{T q}, and for any \tcode{size_t i}
+such that the value of \tcode{i} is less than the length of \tcode{a}.
+\end{note}
 
 \pnum
-Likewise, the expression
-\tcode{\&a[i] != \&b[j]}
-evaluates as
-\tcode{true}
-for any two arrays
-\tcode{a}
-and
-\tcode{b}
-and for any
-\tcode{size_t i}
-and
-\tcode{size_t j}
-such that
-\tcode{i}
-is less than the length of
-\tcode{a}
-and
-\tcode{j}
-is less than the length of
-\tcode{b}.
-This property indicates an absence of aliasing and may be used to
-advantage by optimizing compilers.\footnote{Compilers may take advantage
+\remarks
+The expression \tcode{\&a[i+j] == \&a[i] + j}
```

Comment:
[numeric.defns] bullet 1.9 mean that we can ignore the usual concerns leading to addressof in this case, so I think there is no issue at all.